mava/environment_loop.py

```diff
@@ -575,28 +580,61 @@ def should_run_loop(eval_condtion: Tuple) -> bool:
                         "Running eval loop at executor step: "
                         + f"{self._last_evaluator_run_t}"
                     )
+
             return should_run_loop
 
         episode_count, step_count = 0, 0
 
-        # Currently, we only use intervals for eval loops.
-        environment_loop_schedule = (
-            self._executor._evaluator and self._executor._interval
-        )
-        if environment_loop_schedule:
-            eval_condition = check_count_condition(self._executor._interval)
+        # TODO (Ruan): Checking whether we are using Jax or TF here
+        # this should be removed once we deprecate TF.
+        if hasattr(self._executor, "store"):
+            environment_loop_schedule = self._executor._evaluator and (
+                self._executor.store.evaluation_interval is not None
+            )
+
+            if environment_loop_schedule:
+                eval_condition = check_count_condition(
+                    self._executor.store.evaluation_interval
+                )
+                evaluation_duration = self._executor.store.evaluation_duration
+        else:
+            environment_loop_schedule = (
+                self._executor._evaluator and self._executor._interval
+            )
+            if environment_loop_schedule:
+                eval_condition = check_count_condition(self._executor._interval)
 
         while not should_terminate(episode_count, step_count):
-            if (not environment_loop_schedule) or should_run_loop(eval_condition):
-                result = self.run_episode()
-                episode_count += 1
-                step_count += result["episode_length"]
-                # Log the given results.
-                self._logger.write(result)
+            if (not environment_loop_schedule) or (should_run_loop(eval_condition)):
+                # TODO (Ruan): Remove store check once TF is deprecated.
+                if environment_loop_schedule and hasattr(self._executor, "store"):
+                    # Get first result dictionary
+                    results = self.run_episode()
+                    episode_count += 1
```

Comment:
As discussed, there is a bug where steps and episodes are not updating only on the node itself i.e. on the evaluator node, evaluator steps and evaluator episodes is always 0. Same for executor

Reply:
Fixed. There was a bug in the way the get and set keys were getting defined in the parameter client.